src/main/java/com/rarchives/ripme/ripper/VideoRipper.java

```diff
@@ -44,19 +47,22 @@ public boolean addURLToDownload(URL url, File saveAs) {
         if (Utils.getConfigBoolean("urls_only.save", false)) {
             // Output URL to file
             String urlFile = this.workingDir + File.separator + "urls.txt";
+
+            FileWriter fw = null;
             try {
-                FileWriter fw = new FileWriter(urlFile, true);
+                fw = new FileWriter(urlFile, true);
                 fw.write(url.toExternalForm());
                 fw.write("\n");
-                fw.close();
+
                 RipStatusMessage msg = new RipStatusMessage(STATUS.DOWNLOAD_COMPLETE, urlFile);
                 observer.update(this, msg);
             } catch (IOException e) {
                 logger.error("Error while writing to " + urlFile, e);
                 return false;
+            } finally {
+                close(fw);
```

Comment:
I see why you did this, but maybe there's a slight change to the pattern that would allow you to do fw.close() in the finally block...?